syntax = "proto3";

package document;

service DocumentService {
    rpc cargarDocumento (uploadRequest) returns (uploadResponse);
    rpc descargarDocumento (downloadRequest) returns (downloadResponse);
    rpc listarRecursos (listRequest) returns (listResponse);
}

message uploadRequest {
    string idRemitente = 1;
    string titulo = 2;
    string contenido = 3;
    string cantidadNodos = 4;
    string puertoEscucha = 5;
}
message uploadResponse {
    string idRemitente = 1;
    string mensajeRespuesta = 2;
}


message downloadRequest{
    string idRemitente = 1;
    string titulo = 2;
    string cantidadNodos = 3;
    string puertoEscucha = 4;
}
message downloadResponse{
    string idRemitente = 1;
    string titulo = 2;
    string contenido = 3;
}

message listRequest{
    string idRemitente = 1;
    string idDestinatario = 2;
    string cantidadNodos = 3;
    string puertoEscucha = 4;
}

message Documento{
    string titulo = 1;
    string contenido = 2;
}

message listResponse{
    string idRemitente = 1;
    repeated Documento documentos = 2;
}
